src/main/proto/netflix/titus/agent.proto

```diff
@@ -371,3 +395,11 @@ message TaskIdentityStringDocument {
     // (Required) Signature produced by signing identity bytes.
     optional CertificateStringSignature signature = 2;
 }
+
+message TaskPodIdentityDocument {
+    // (Required) Serialized form of a TaskIdentity protobuf.
+    optional bytes podIdentity = 1;
```

Comment:
@rgulewich @JackOfMostTrades do we need this second doc too?

It looks like it has the Sig and stuff, but is there any particular reason why we make two docs like this?

```
	taskIdent := ms.generateTaskIdentity()
	identData, err := proto.Marshal(taskIdent)
	sig, err := ms.signer.Sign(identData)
	taskIdentDoc := &titus.TaskIdentityDocument{
		Identity:  identData,
		Signature: sig,
	}
```

I guess it is because we have to sign *bytes* ?
So therefore we need two `messages`, one that has the data, which can be turned into bytes, then we can sign those bytes into the second message.

If we do have to do it this way (as opposed to having just one message that has the pod + ip + whatever + sig (signing the above concatinated?), then I'll merge and implement.

If there isn't a reason to do it in 2 steps anymore (historical, idk), then let me know now and we can simplify.

Reply:
Yea, exactly what you said, signatures operate on _bytes_ and on the other end we will verify some _bytes_ . To the extent that you wanted to encode the combined result differently, (e.g. a concatenation of the bytes + sig) that's fine, but it would be best to prefix some sort of version / magic number to that to leave ourselves some room to update the structure later, and at the point that you're trying to implement this sort of serialization / parsing of structured data, why do it all manually when proto gives you a way to define, parse, and serialize structured data. ;)